lib/3scale/backend/storage_async/pipeline.rb

```diff
@@ -73,6 +80,37 @@ def run(redis_async_client)
             end
           end
         end
+
+        private
+
+        def collect_responses(redis_async_client)
+          async_pipe = redis_async_client.pipeline
+          @commands.each do |command|
+            async_pipe.write_request(*command)
+          end
+
+          # Redis returns an answer for each of the commands sent in the
+          # pipeline. But in order to keep compatibility with redis-rb, here, if
+          # there is an error in any of the commands of the pipeline we will
+          # raise an error (the first one that occurred).
+
+          first_err = nil
+
+          res = @commands.size.times.map do
+            begin
+              async_pipe.read_response
+            rescue ::Protocol::Redis::ServerError => e
+              first_err ||= e
```

Comment:
Why not immediately raise here? Or is it related to freeing Redis resources?

Reply:
I see the original code was like this. Fine with me to stay as is if you don't want to bother now figuring this out.

Reply:
Yeah, the comment above explains it, it's to match `redis-rb` behavior.

```ruby
# Redis returns an answer for each of the commands sent in the
# pipeline. But in order to keep compatibility with redis-rb, here, if
# there is an error in any of the commands of the pipeline we will
# raise an error (the first one that occurred).
```

Reply:
well, if we immediately raise, then the comment will still hold true. Why go over the remaining results if we are going to raise anyway? The only reaon I can think of is in case database resources will not be freed.

Reply:
I don't know the implementation details of the pipeline in `redis-rb`, maybe this can be done in some other way, but this is how it was before https://github.com/3scale/apisonator/commit/730442ef15a17daebfb3e584a582c965bb4b86df and it works so I didn't touch it.